runtime/strategies/map-slots.js

```diff
@@ -26,16 +26,20 @@ export default class MapSlots extends Strategy {
         }
 
         let selectedSlots = MapSlots.findAllSlotCandidates(slotConnection, arc);
+
+        //  ResolveRecipe handles a one slot case.
         if (selectedSlots.length < 2) {
           return;
         }
 
-        let selectedSlot = selectedSlots[0]; // TODO: return combinatorial results?
-
-        return (recipe, slotConnection) => {
-          MapSlots.connectSlotConnection(slotConnection, selectedSlot);
-          return 1;
-        };
+        return selectedSlots
+            // If there are multiple options, filter out context slots.
+            // TODO: return full combinatorial results?
```

Comment:
what tests do you need to rework in order to not filter out the context slots? i'd rather we don't remove them, because the logic of doing it is not really clear to me, and I'd rather identify cases when it happens sooner rather than later.

one concern that i have is demo-flow-test is disabled (i'll work on re-enabling it tomorrow), and i wonder what this change would have done to it. 

otherwise, looks good.